src/core/a-entity.js

```diff
@@ -435,14 +435,28 @@ var proto = Object.create(ANode.prototype, {
       if (isDefault || isMixedIn) { return; }
 
       component = this.components[name];
+
+      if (!component) { return; }
+
+      // Wait for component to initialize.
+      if (!component.initialized) {
```

Comment:
Asynchrony might bite you. What happens if you do:
````javascript
myMethod () {
this.setAttribute('my-component', { });
this.removeAttribute('my-component');
this.setAttribute('my-component', {});
}````
I believe the component will be removed if the component was not initialized in the first call
```